azurerm/internal/services/sentinel/sentinel_automation_rule_resource_test.go

```diff
@@ -0,0 +1,359 @@
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccSentinelAutomationRule_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_sentinel_automation_rule", "test")
+	r := SentinelAutomationRuleResource{uuid: uuid.New().String()}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccSentinelAutomationRule_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_sentinel_automation_rule", "test")
+	r := SentinelAutomationRuleResource{uuid: uuid.New().String()}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (r SentinelAutomationRuleResource) Exists(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	client := clients.Sentinel.AutomationRulesClient
+
+	id, err := parse.AutomationRuleID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp, err := client.Get(ctx, id.ResourceGroup, sentinel.OperationalInsightsResourceProvider, id.WorkspaceName, id.Name); err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	return utils.Bool(true), nil
+}
+
+func (r SentinelAutomationRuleResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_sentinel_automation_rule" "test" {
+  name                       = "%s"
+  log_analytics_workspace_id = azurerm_log_analytics_solution.sentinel.workspace_resource_id
+  display_name               = "acctest-SentinelAutoRule-%d"
+  order                      = 1
+  condition {
+    property = "IncidentTitle"
+    operator = "Contains"
+    values   = ["a", "b"]
+  }
+
+  action_incident {
+    order  = 1
+    status = "Active"
+  }
+}
+`, template, r.uuid, data.RandomInteger)
+}
+
+func (r SentinelAutomationRuleResource) complete(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_role_assignment" "sentinel" {
+  scope                = azurerm_resource_group.test.id
+  role_definition_name = "Azure Sentinel Automation Contributor"
+  principal_id         = "df410494-9bdd-4bbe-997f-51bab37e3d91"
```

Comment:
I think i mightve mentioned this but the object ID of Azure Security insights changes per tenant, this either needs to be done manually, make it a required env variable or we need to pull in the azure ad provider to figure this out dynamically

Reply:
```
data "azurerm_client_config" "current" {}

provider "azuread" {
  }

data "azuread_service_principal" "securityinsights" {
	display_name = "Azure Security Insights"
  }

resource "azurerm_role_assignment" "sentinel" {
  scope                = azurerm_resource_group.test.id
  role_definition_name = "Azure Sentinel Automation Contributor"
  principal_id         = data.azuread_service_principal.securityinsights.object_id
}
```
This will do it. Works for me. Note for service principles 
Application.Read.All from Azure Active Directory Graph must be assigned within API Permissions and granted admin consent due to how AzureAD Provider works. We should probably warn this somewhere in the test if we do take this approach. 

```
=== RUN   TestAccSentinelAutomationRule_basic
=== PAUSE TestAccSentinelAutomationRule_basic
=== RUN   TestAccSentinelAutomationRule_complete
=== PAUSE TestAccSentinelAutomationRule_complete
=== RUN   TestAccSentinelAutomationRule_update
=== PAUSE TestAccSentinelAutomationRule_update
=== RUN   TestAccSentinelAutomationRule_requiresImport
=== PAUSE TestAccSentinelAutomationRule_requiresImport
=== CONT  TestAccSentinelAutomationRule_basic
=== CONT  TestAccSentinelAutomationRule_requiresImport
=== CONT  TestAccSentinelAutomationRule_update
=== CONT  TestAccSentinelAutomationRule_complete
--- PASS: TestAccSentinelAutomationRule_basic (263.35s)
--- PASS: TestAccSentinelAutomationRule_requiresImport (342.66s)
--- PASS: TestAccSentinelAutomationRule_complete (417.94s)
--- PASS: TestAccSentinelAutomationRule_update (718.39s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/sentinel    718.497s
```

Reply:
@kaovd, this is the reason why I make it draft. I didn't figure out what SP that ID corresponds to. Thank you so much!

Reply:
Regarding the order clobber, I noticed the portal behavior. Whilst I feel it isn't something we shall follow on client side as the API allows clobbered order, otherwise, it will be hard to import existing automation rule which has clobbered order into Terraform?

Reply:
Yeah no problem can leave it then, although situation for that feels unlikely.